docs/Guidelines(8th Edition).xml

```diff
@@ -1,251 +1,251 @@
-	<guideline key="Ch22_13460e8" severity="DO NOT" section="Coding" subsection="Synchronization">DO NOT request exclusive ownership of the same two or more synchronization targets in different orders.</guideline>
-	<guideline key="Ch22_025dd17" severity="DO" section="Coding" subsection="Threads">DO ensure that code that concurrently holds multiple locks always acquires them in the same order.</guideline>
-	<guideline key="Ch22_52cbe24" severity="DO" section="Coding" subsection="Synchronization">DO encapsulate mutable static data in public APIs with synchronization logic.</guideline>
-	<guideline key="Ch22_14c2494" severity="AVOID" section="Coding" subsection="Synchronization">"AVOID synchronization on simple reading or writing of values no bigger than a native (pointer-size) integer, as such operations are automatically atomic."</guideline>
-	<guideline key="Ch23_ac15501" severity="DO" section="Coding" subsection="Methods">"DO create public managed wrappers around unmanaged methods that use the conventions of managed code, such as structured exception handling."</guideline>
-	<guideline key="Ch23_1cbd746" severity="DO NOT" section="Coding" subsection="Classes">DO NOT unnecessarily replicate existing managed classes that already perform the function of the unmanaged API.</guideline>
-	<guideline key="Ch23_a0fe82e" severity="DO" section="Coding" subsection="Methods">DO declare extern methods as private or internal.</guideline>
-	<guideline key="Ch23_3189bfc" severity="DO" section="Coding" subsection="Methods">"DO provide public wrapper methods that use managed conventions such as structured exception handling, use of enums for special values, and so on."</guideline>
-	<guideline key="Ch23_4c7fa17" severity="DO" section="Coding" subsection="Parameters">DO simplify the wrapper methods by choosing default values for unnecessary parameters.</guideline>
-	<guideline key="Ch23_3969d9e" severity="DO" section="Coding" subsection="Exceptions">DO use the SetLastErrorAttribute on Windows to turn APIs that use SetLastError error codes into methods that throw Win32Exception.</guideline>
-	<guideline key="Ch23_193ed1d" severity="DO" section="Coding" subsection="Methods">DO extend SafeHandle or implement IDisposable and create a finalizer to ensure that unmanaged resources can be cleaned up effectively.</guideline>
-	<guideline key="Ch23_e606562" severity="DO" section="Coding" subsection="Miscellaneous">DO use delegate types that match the signature of the desired method when an unmanaged API requires a function pointer.</guideline>
-	<guideline key="Ch23_d560942" severity="DO" section="Coding" subsection="Miscellaneous">DO use ref parameters rather than pointer types when possible.</guideline>
-</Guidelines>
+  <guideline key="Ch03_2de296a" severity="CONSIDER" section="Coding" subsection="Arrays">CONSIDER checking the array length before indexing into an array rather than assuming the length.</guideline>
```

Comment:
Any reason for re-ordering the items in the XML? The markdown generator doesn't care about the order of the XML items and with the order switched it makes it difficult to determine the actual changes in here.

Reply:
The reason was just so it was easy to combine topics for the md. With it all spread around it would of been extremely difficult assigning different sections to other subsections efficiently and accurately. Since I moved guidelines out of everything except coding section, this made sense to visualize what steps I was taking and which guidelines were going into which section. @Keboo